django/template/loaders/cached.py

```diff
@@ -26,12 +26,25 @@ def __init__(self, engine, loaders):
     def get_contents(self, origin):
         return origin.loader.get_contents(origin)
 
+    def _cache_exception(self, exc):
+        if isinstance(exc, type) and issubclass(exc, TemplateDoesNotExist):
+            return exc
+        elif isinstance(exc, TemplateDoesNotExist):
+            if self.engine.debug:
+                return exc.__class__(*exc.args, tried=exc.tried)
+            return exc.__class__
```

Comment:
For memory reasons, we want to cache just the raw exception class, so that the overhead for each missing template is just a single reference to an existing object, rather than a unique object for each template.

Ultimately, the exception that propagates upward in the code is [this one](https://github.com/django/django/blob/0e80ac467293c659588769ec0587a5bf5f084b49/django/template/engine.py#L169), and that contains the template name.